src/ServicePulse.Host/app/js/views/message/controller.js

```diff
@@ -191,7 +192,7 @@
                 dnsName = dnsName.replace("http://", "");
             }
 
-            $window.open("si://" + dnsName + "?search=" + messageId);
+            $window.open("si://" + dnsName + "?search=" + messageId + "&endpointname=" + endpointName);
```

Comment:
Don't these values need escaping or is that done automatically by `$window.open`?

Reply:
it's done automatically IIRC, also I don't think there can be many strange chars in an endpoint name.